### PR TITLE
bumped dependency versions

### PR DESCRIPTION
**Pull request by michaelkamprath**
Bumped various dependency versions

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "avr-progmem"
-version = "0.3.3"
```
Comment (Cryptjar) [inline Cargo.toml:3]: Upgrading `ufmt` is a breaking change, so we need to bump the version to 0.4.0. Thus, the PR should also target the new `v0.4` branch instead.

File: Cargo.toml [hunk 2]
```diff
@@ -82,4 +82,4 @@ rev = "81edfcd908be8d3b2cd763cf2dc76aedbb22286b"
 # Fixes formatting of u32, however, it causes a link error due to u32-div
 # Which in turn is fixed (in config.toml) by:
 #     build-std-features = ["compiler-builtins-mangled-names"]
-ufmt = { git = "https://github.com/mrk-its/ufmt.git", rev = "e6e574625e89bce9a5cb907eae43bfd45672f768" }
```
Comment (Cryptjar) [inline Cargo.toml:85]: If `ufmt` v0.2 fixes the `u32` formatting, we no longer need the patch section.

File: Cargo.toml (hunk 1 above)
Comment (michaelkamprath) [reply Cargo.toml:3]: Updated

File: Cargo.toml (hunk 2 above)
Comment (michaelkamprath) [reply Cargo.toml:85]: Removed

**Comment by Cryptjar** [comment]
Looks good

**Comment by Cryptjar** [comment]
Released as `v0.4.0`

**Pull request merged**